servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java

```diff
@@ -189,9 +189,9 @@ protected Object buildBody() {
     private Argument<?> resolveBodyType() {
         RouteMatch<?> route = this.getAttribute(HttpAttributes.ROUTE_MATCH, RouteMatch.class).orElse(null);
         if (route != null) {
-            Argument<?> bodyType = route.getRouteInfo().getFullBodyArgument()
+            Argument<?> bodyType = route.getRouteInfo().getFullRequestBodyType()
                 /*
-                The getBodyArgument() method returns arguments for functions where it is
```

Comment:
not sure this filter is necessary anymore

Reply:
I removed the filter. Please confirm I did what you expected.